File: replace_paco.py
import re

legacy_path = r'D:\New road travels\redbus-clone\cms-platform\apps\api\src\modules\payment\paco-gateway.service.ts'
new_path = r'd:\distributed bus management system\bus-booking-backend\src\payments\providers\paco.provider.ts'

with open(legacy_path, 'r', encoding='utf-8') as f:
    legacy_content = f.read()
    
with open(new_path, 'r', encoding='utf-8') as f:
    new_content = f.read()

def extract_val(prop_name):
    # Match something like: private readonly paymentEndpoint = process.env.PACO_ENDPOINT || 'https://core.demo-paco.2c2p.com/';
    pattern = r"private readonly " + prop_name + r"\s*=\s*process\.env\.[A-Z_0-9]+\s*\|\|\s*'([^']+)'"
    match = re.search(pattern, legacy_content)
    if match:
        return match.group(1)
    return None

def extract_val_key(key_name, env_name):
    # Match something like: NPR: process.env.PACO_API_KEY_NPR || process.env.PACO_API_KEY || '65805a1636c74b8e8ac81a991da80be4',
    pattern = key_name + r": process\.env\." + env_name + r"(?:\s*\|\|\s*process\.env\.[A-Z_0-9]+)?\s*\|\|\s*'([^']+)'"
    match = re.search(pattern, legacy_content)
    if match:
        return match.group(1)
    return None

def replace_val(prop_name, new_val):
    global new_content
    pattern = r"(private readonly " + prop_name + r"\s*=\s*process\.env\.[A-Z_0-9]+\s*\|\|\s*')[^']+'"
    new_content = re.sub(pattern, r"\g<1>" + new_val.replace('\\', '\\\\') + "'", new_content)

def replace_val_key(key_name, env_name, new_val):
    global new_content
    pattern = r"(" + key_name + r": process\.env\." + env_name + r"(?:\s*\|\|\s*process\.env\.[A-Z_0-9]+)?\s*\|\|\s*')[^']+'"
    new_content = re.sub(pattern, r"\g<1>" + new_val.replace('\\', '\\\\') + "'", new_content)

props = [
    'paymentEndpoint',
    'merchantId',
    'encryptionKeyId',
    'merchantSigningPrivateKeyB64',
    'pacoEncryptionPublicKeyB64',
    'pacoSigningPublicKeyB64',
    'merchantDecryptionPrivateKeyB64'
]

for p in props:
    v = extract_val(p)
    if v:
        replace_val(p, v)

v_npr = extract_val_key('NPR', 'PACO_API_KEY_NPR')
if v_npr:
    replace_val_key('NPR', 'PACO_API_KEY_NPR', v_npr)

v_usd = extract_val_key('USD', 'PACO_API_KEY_USD')
if v_usd:
    replace_val_key('USD', 'PACO_API_KEY_USD', v_usd)

with open(new_path, 'w', encoding='utf-8') as f:
    f.write(new_content)
    
print("Successfully replaced PACO keys")

File: replace_paco.js
const fs = require('fs');
const path = require('path');
const legacyPath = 'D:\\New road travels\\redbus-clone\\cms-platform\\apps\\api\\src\\modules\\payment\\paco-gateway.service.ts';
const newPath = 'd:\\distributed bus management system\\bus-booking-backend\\src\\payments\\providers\\paco.provider.ts';

const legacyContent = fs.readFileSync(legacyPath, 'utf8');
let newContent = fs.readFileSync(newPath, 'utf8');

// Use string operations to extract keys
function extractVal(propName) {
  let idx = legacyContent.indexOf('private readonly ' + propName);
  if(idx === -1) return null;
  let idx2 = legacyContent.indexOf("'", idx);
  let idx3 = legacyContent.indexOf("'", idx2 + 1);
  return legacyContent.substring(idx2 + 1, idx3);
}

function extractApiKey(prefix) {
  let idx = legacyContent.indexOf(prefix);
  if(idx === -1) return null;
  let idx2 = legacyContent.indexOf("'", idx);
  let idx3 = legacyContent.indexOf("'", idx2 + 1);
  return legacyContent.substring(idx2 + 1, idx3);
}

const paymentEndpoint = extractVal('paymentEndpoint');
const merchantId = extractVal('merchantId');
const encryptionKeyId = extractVal('encryptionKeyId');
const merchantSigningPrivateKeyB64 = extractVal('merchantSigningPrivateKeyB64');
const pacoEncryptionPublicKeyB64 = extractVal('pacoEncryptionPublicKeyB64');
const pacoSigningPublicKeyB64 = extractVal('pacoSigningPublicKeyB64');
const merchantDecryptionPrivateKeyB64 = extractVal('merchantDecryptionPrivateKeyB64');

const apiKeyNPR = extractApiKey('NPR: process.env.PACO_API_KEY_NPR');
const apiKeyUSD = extractApiKey('USD: process.env.PACO_API_KEY_USD');

function replaceVal(propName, newVal) {
  let searchStr = 'private readonly ' + propName;
  let idx = newContent.indexOf(searchStr);
  if(idx === -1) return;
  let idx2 = newContent.indexOf("'", idx);
  let idx3 = newContent.indexOf("'", idx2 + 1);
  
  newContent = newContent.substring(0, idx2 + 1) + newVal + newContent.substring(idx3);
}

function replaceApiKey(prefix, newVal) {
  let idx = newContent.indexOf(prefix);
  if(idx === -1) return;
  let idx2 = newContent.indexOf("'", idx);
  let idx3 = newContent.indexOf("'", idx2 + 1);
  
  newContent = newContent.substring(0, idx2 + 1) + newVal + newContent.substring(idx3);
}

replaceVal('paymentEndpoint', paymentEndpoint);
replaceVal('merchantId', merchantId);
replaceVal('encryptionKeyId', encryptionKeyId);
replaceVal('merchantSigningPrivateKeyB64', merchantSigningPrivateKeyB64);
replaceVal('pacoEncryptionPublicKeyB64', pacoEncryptionPublicKeyB64);
replaceVal('pacoSigningPublicKeyB64', pacoSigningPublicKeyB64);
replaceVal('merchantDecryptionPrivateKeyB64', merchantDecryptionPrivateKeyB64);

replaceApiKey('NPR: process.env.PACO_API_KEY_NPR', apiKeyNPR);
replaceApiKey('USD: process.env.PACO_API_KEY_USD', apiKeyUSD);

fs.writeFileSync(newPath, newContent);
console.log('Successfully updated PACO keys in paco.provider.ts');
